MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs

```diff
@@ -122,55 +142,22 @@ private static void HandleContinue(GameSession session, PacketReader packet)
             ScriptMetadata scriptMetadata = npcTalk.IsQuest ? ScriptMetadataStorage.GetQuestScriptMetadata(npcTalk.QuestId) : ScriptMetadataStorage.GetNpcScriptMetadata(npcTalk.Npc.Id);
             ResponseType responseType = npcTalk.IsQuest ? ResponseType.Quest : ResponseType.Dialog;
 
-            if (npcTalk.ScriptId != 0 && scriptMetadata.Options.First(x => x.Id == npcTalk.ScriptId).AmountContent <= npcTalk.ContentIndex && scriptMetadata.Options.First(x => x.Id == npcTalk.ScriptId).Goto.Count == 0)
+            if (npcTalk.ScriptId != 0
+                && scriptMetadata.Options.First(x => x.Id == npcTalk.ScriptId).AmountContent <= npcTalk.ContentIndex
```

Comment:
can you cache `scriptMetadata.Options.First(x => x.Id == npcTalk.ScriptId)` so you don't need to query it twice?